chains/evm/cli/bridge/register-resource.go

```diff
@@ -22,62 +21,74 @@ var registerResourceCmd = &cobra.Command{
 		txFabric := evmtransaction.NewTransaction
 		return RegisterResourceCmd(cmd, args, txFabric)
 	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		err := validateRegisterResourceFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = processRegisterResourceFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
 }
 
-func BindRegisterResourceCmdFlags(cli *cobra.Command) {
-	cli.Flags().String("handler", "", "handler contract address")
-	cli.Flags().String("bridge", "", "bridge contract address")
-	cli.Flags().String("target", "", "contract address to be registered")
-	cli.Flags().String("resourceId", "", "resource ID to be registered")
+func BindRegisterResourceCmdFlags() {
+	registerResourceCmd.Flags().StringVarP(&Handler, "handler", "h", "", "handler contract address")
+	registerResourceCmd.Flags().StringVarP(&Bridge, "bridge", "b", "", "bridge contract address")
+	registerResourceCmd.Flags().StringVarP(&Target, "target", "t", "", "contract address to be registered")
+	registerResourceCmd.Flags().StringVarP(&ResourceID, "resourceId", "rID", "", "resource ID to be registered")
+	flags.CheckRequiredFlags(registerResourceCmd, "handler", "bridge", "target", "resourceId")
 }
 
 func init() {
-	BindRegisterResourceCmdFlags(registerResourceCmd)
+	BindRegisterResourceCmdFlags()
+}
+
+func validateRegisterResourceFlags(cmd *cobra.Command, args []string) error {
+	if !common.IsHexAddress(Handler) {
+		return fmt.Errorf("invalid handler address %s", Handler)
+	}
+	if !common.IsHexAddress(Target) {
+		return fmt.Errorf("invalid target address %s", Target)
+	}
+	if !common.IsHexAddress(Bridge) {
+		return fmt.Errorf("invalid bridge address %s", Bridge)
+	}
+	return nil
+}
+
+func processRegisterResourceFlags(cmd *cobra.Command, args []string) error {
+	var err error
+	handlerAddr = common.HexToAddress(Handler)
+	targetContractAddr = common.HexToAddress(Target)
+	bridgeAddr = common.HexToAddress(Bridge)
+
+	resourceIdBytesArr, err = flags.ProcessResourceID(ResourceID)
+	if err != nil {
```

Comment:
There is no need for this check, you can just return `err` as it will be `nil` if everything is ok